packages/remix-dev/compiler/plugins/serverNativeNodeModules.ts

```diff
@@ -0,0 +1,32 @@
+import type { Plugin } from "esbuild";
+
+// See https://github.com/evanw/esbuild/issues/1051#issuecomment-806325487
+export const serverNativeNodeModulesPlugin: Plugin = {
+  name: 'native-node-modules',
+  setup(build) {
+    // If a ".node" file is imported within a module in the "file" namespace, resolve
+    // it to an absolute path and put it into the "node-file" virtual namespace.
+    build.onResolve({ filter: /\.node$/, namespace: 'file' }, args => ({
+      path: require.resolve(args.path, { paths: [args.resolveDir] }),
+      namespace: 'node-file',
+    }))
+
+    // Files in the "node-file" virtual namespace call "require()" on the
+    // path from esbuild of the ".node" file in the output directory.
+    build.onLoad({ filter: /.*/, namespace: 'node-file' }, args => ({
+      contents: `
+        import path from ${JSON.stringify(args.path)}
+        try { module.exports = require(path) }
```

Comment:
@jacob-ebey's comment (https://github.com/remix-run/remix/pull/3263#discussion_r910177928)
> This will break node ESM. Is it possible to load via `import` statements?

@benwis' reaction (https://github.com/remix-run/remix/pull/3263#discussion_r910203192)
> The trouble is getting the absolute path and then loading it. I can't use await import() because I can't have top level await. Unless there's a way to get the absolute path and the compiled path without importing it, I don't see how I resolve this